### PR TITLE
validate manifests size

### PR DESCRIPTION
**Pull request by zhiweiyin318**
Signed-off-by: Zhiwei Yin <zyin@redhat.com>

### CODE REVIEW

File: pkg/webhook/manifestwork_webhook.go [hunk 1]
```diff
@@ -16,7 +16,8 @@ import (
 	"k8s.io/klog/v2"
 )
 
-var manifestLimit = 10
+// ManifestLimit is the max size of manifests data which is 50k bytes.
+const ManifestLimit = 50 * 1000
```
Comment (qiujian16) [inline pkg/webhook/manifestwork_webhook.go]: 50 * 1024

Comment (zhiweiyin318) [reply pkg/webhook/manifestwork_webhook.go]: done

**Comment by qiujian16** [comment]
/approve
/lgtm
/hold

Let's merge it after this sprint.

File: pkg/webhook/manifestwork_webhook.go [hunk 2]
```diff
@@ -90,8 +91,13 @@ func (a *ManifestWorkAdmissionHook) validateManifestWorkObj(requestObj runtime.R
...
 
-	if len(work.Spec.Workload.Manifests) > manifestLimit {
-		return fmt.Errorf("number of manifests should not be larger than %d", manifestLimit)
+	totalSize := 0
+	for _, manifest := range work.Spec.Workload.Manifests {
+		totalSize = totalSize + manifest.Size()
+	}
+
+	if totalSize > ManifestLimit {
+		return fmt.Errorf("the size of manifests should not be larger than 50k bytes")
```
Comment (qiujian16) [inline pkg/webhook/manifestwork_webhook.go]: it is better to also  output the  actual size here.

Comment (zhiweiyin318) [reply pkg/webhook/manifestwork_webhook.go]: done

**Pull request merged**